Car_owners/tests/test_views.py

```diff
@@ -0,0 +1,269 @@
+import datetime
+import pytest
+from rest_framework import status
+from rest_framework.test import APIClient
+from application.views import OwnerViewSet, CarViewSet
+
+
+@pytest.fixture
+def api_client() -> APIClient:
+    yield APIClient()
+
+
+@pytest.fixture
+def owner_data() -> dict[str, str]:
+    owner_data = {"name": "Andrzej", "surname": "Starczyk", "phone": "575849675"}
+    return owner_data
+
+
+@pytest.fixture
+def car_data(api_client: APIClient, owner_data: dict[str, str]) -> dict[str, str | int]:
+    response_create_owner = api_client.post(
+        "/app/owners/", data=owner_data, format="json"
+    )
+    owner_id = response_create_owner.data["id"]
+    car_data = {
+        "brand": "Ford",
+        "model": "Mondeo",
+        "production_date": "2023-01-01",
+        "owner": owner_id,
+    }
+
+    return car_data
+
+
+@pytest.mark.django_db
+def test_create_and_get_owner(
+    api_client: APIClient, owner_data: dict[str, str]
+) -> None:
+    # create owner
+    response_create_owner = api_client.post(
+        "/app/owners/", data=owner_data, format="json"
+    )
+    owner_id = response_create_owner.data["id"]
+    assert response_create_owner.status_code == status.HTTP_201_CREATED
+    assert all(
+        [
+            response_create_owner.data[key] == owner_data[key]
+            for key in owner_data.keys()
+        ]
+    )
+
+    # get owner
+    response_get_owner = api_client.get(f"/app/owners/{owner_id}/", format="json")
+    assert response_get_owner.status_code == status.HTTP_200_OK
+    assert all(
+        [response_get_owner.data[key] == owner_data[key] for key in owner_data.keys()]
+    )
+
+
+@pytest.mark.django_db
+def test_patch_owner(api_client: APIClient, owner_data: dict[str, str]) -> None:
+    # create owner
+    response_create_owner = api_client.post(
+        "/app/owners/", data=owner_data, format="json"
+    )
+    owner_id = response_create_owner.data["id"]
+    assert response_create_owner.status_code == status.HTTP_201_CREATED
+    assert all(
+        [
+            response_create_owner.data[key] == owner_data[key]
+            for key in owner_data.keys()
+        ]
+    )
+
+    # update owner
+    owner_data["name"] = "Krzysztof"
```

Comment:
create a new object, `new_owner_data` and use this fresh one to update - real scenario